app/src/test/java/com/codesoom/assignment/domain/UserTest.java

```diff
@@ -0,0 +1,73 @@
+package com.codesoom.assignment.domain;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+@DisplayName("User 클래스")
+class UserTest {
+
+    private User user;
+
+    @BeforeEach
+    void setUp(){
+        user = User.builder()
+                .id(1L)
+                .name("김철수")
+                .email("abc@exmaple.com")
+                .pw("1234")
+                .build();
+    }
+
+
+    @Nested
+    @DisplayName("builder는")
+    class Describe_creationWithBuilder{
+
+        @Test
+        @DisplayName("User를 리턴한다.")
+        void it_returns_user(){
+            assertThat(user.getId()).isEqualTo(1L);
+            assertThat(user.getName()).isEqualTo("김철수");
+            assertThat(user.getEmail()).isEqualTo("abc@exmaple.com");
+            assertThat(user.getPw()).isEqualTo("1234");
+        }
+    }
+
+    @Nested
+    @DisplayName("changeWith 메소드는")
+    class Describe_changeWith{
+
+        @Nested
+        @DisplayName("User 객체가 주어지면")
+        class Context_with_a_user{
+
+            @BeforeEach
+            void setUp(){
```

Comment:
user를 공용으로 사용하지 않는 것이 좋습니다. `Describe`와 `Context`안쪽은 하나의 실험실이라 생각하고 테스트를 작성하세요. 실험실들끼리 실험 대상을 공유해서 쓰면 실험 결과도 엉뚱하게 나올 수 있겠죠.
```suggestion
        @Nested
        @DisplayName("User 객체가 주어지면")
        class Context_with_a_user{

            User givenUser;

            @BeforeEach
            void setUp(){
```

Reply:
실험실끼리 같이 쓰다보면 엉뚱한 결과를 얻을 가능성이 존재하게 되겠네요!
분리하겠습니다ㅎㅎ
다만 실험대상을 분리하다보니깐 기존에 공유해서 초기화하는 부분도 분리하게되어서
아래처럼 초기화하는 코드가 2개의 실험실에서 각각 사용되는데요.
```java
                givenUser = User.builder()
                        .id(1L)
                        .name("김철수")
                        .email("abc@exmaple.com")
                        .pw("1234")
                        .build();
```
만약 테스트가 많아지고 각 테스트마다 위처럼 철수로 일관되게 초기화시키면 코드 중복이 많이 일어날거 같습니다. 그래서
1. 초기화 상태를 명확하게 보여주니깐 그대로 두는게 좋을지
2. 똑같은 상태로 초기화시켜주고 코드중복이 많으니깐 초기화하는 메서드를 따로 빼서 사용하는게 좋을지
어떤게 좋을지 궁금합니다.
저는 철수로 초기화하는 테스트가 3개 이상정도 되면 2번 의견처럼 초기화 메서드를 따로 구현해서 사용하면 좋을거 같다고 생각이 들긴합니다.

Reply:
그것도 선택입니다. 중복을 적절히 없앨 것인지, 아니면 명시적으로 각 테스트를 확실히 격리할 것인지. 저는 테스트에 대해서는 중복이 좀 있더라도 확실히 분리하는 걸 선호하는 편이긴 합니다. 그렇게 안 했다가 큰 코 다친 적이 있거든요. 회사에서 작성하는 테스트 코드는 정말 주의깊고 꼼꼼하고 확실히 테스트하는 것이 좋습니다. 테스트에 빠뜨린 코드에서 버그가 발생해 실수로 수십억원을 날려먹는 경우도 있을 수 있어요.

Reply:
수십억원이라니 아찔하네요.. 저도 확실하게 분리하는 쪽으로 짜고 싶어지네요. 감사합니다!!